sys/include/net/gnrc/tcp/config.h

```diff
@@ -185,6 +185,26 @@ extern "C" {
 #ifndef CONFIG_GNRC_TCP_EVENTLOOP_MSG_QUEUE_SIZE_EXP
 #define CONFIG_GNRC_TCP_EVENTLOOP_MSG_QUEUE_SIZE_EXP (3U)
 #endif
+
+/**
+ * @brief Enable experimental feature "dynamic msl". Disabled by default.
+ * @note This features calculates the MSL based by multiplying the latest
+ *       retransmission timeout value with
+ *       CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_RTO_MUL. This leads to much
+ *       faster return times on gnrc_tcp_close.
+ */
+#ifndef CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN
+#define CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN (0U)
```

Comment:
```suggestion
#define CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN 0
```

Reply:
So `IS_ACTIVE` macro works

Reply:
Maybe also an `@experimental` note on the docs 😉

Reply:
Remove the parens 👀 otherwise if `CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN` is not defined by Kconfig compilation will fail, e.g.:

```
In file included from /home/jeandudey/Dev/RIOT-OS/RIOT/core/include/panic.h:25:0,
                 from /home/jeandudey/Dev/RIOT-OS/RIOT/core/include/assert.h:25,
                 from /home/jeandudey/Dev/RIOT-OS/RIOT/sys/include/utlist.h:43,
                 from /home/jeandudey/Dev/RIOT-OS/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c:20:
/home/jeandudey/Dev/RIOT-OS/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c: In function ‘_restart_timewait_timer’:
/home/jeandudey/Dev/RIOT-OS/RIOT/core/include/kernel_defines.h:196:52: error: pasting "__PREFIX_WHEN_" and "(" does not give a valid preprocessing token
 #define __is_active(val)              ___is_active(__PREFIX_WHEN_##val)
                                                    ^
/home/jeandudey/Dev/RIOT-OS/RIOT/core/include/kernel_defines.h:177:26: note: in expansion of macro ‘__is_active’
 #define IS_ACTIVE(macro) __is_active(macro)
                          ^~~~~~~~~~~
/home/jeandudey/Dev/RIOT-OS/RIOT/sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c:121:9: note: in expansion of macro ‘IS_ACTIVE’
     if (IS_ACTIVE(CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN)) {
```

To test it:

```diff
diff --git a/sys/include/net/gnrc/tcp/config.h b/sys/include/net/gnrc/tcp/config.h
index 6fc638cb1d..91b5c4e918 100644
--- a/sys/include/net/gnrc/tcp/config.h
+++ b/sys/include/net/gnrc/tcp/config.h
@@ -194,6 +194,7 @@ extern "C" {
  *       CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_RTO_MUL. This leads to much
  *       faster return times on gnrc_tcp_close.
  */
+#undef CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN
 #ifndef CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN
 #define CONFIG_GNRC_TCP_EXPERIMENTAL_DYN_MSL_EN (0)
 #endif
```